intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -138,37 +138,44 @@ def convert_float(value):
         return float(value)
 
 
-def convert_hostname_and_url(value, row):
+def convert_http_host_and_url(value, row):
     """
-    URLs are split into hostname and path, we can also guess the protocol here.
-    but only guess if the protocol is in a set of known good values.
+    URLs are split into hostname and path. The column names differ in reports.
+    Compromised-Website: http_host, url
+    Drone: cc_dns, url
+    IPv6-Sinkhole-HTTP-Drone: http_host, http_url
+    Microsoft-Sinkhole: http_host, url
+    Sinkhole-HTTP-Drone: http_host, url
+    With some reports, url/http_url holds only the path, with others the full HTTP request.
     """
-    if row['application'] in ['http', 'https', 'irc']:
-        if row['hostname'] and row['url']:
-            url = row['url'] if row['url'].startswith('/') else '/' + row['url']
-            return row['application'] + '://' + row['hostname'] + url
+    hostname = ""
+    if "cc_dns" in row:
+        if row['cc_dns']:
+            hostname = row['cc_dns']
+    elif "http_host" in row:
+        if row['http_host']:
+            hostname = row['http_host']
+
+    path = ""
+    if "url" in row:
+        if row['url']:
+            path = row['url']
+    elif "http_url" in row:
+        if row['http_url']:
+            path = row['http_url']
 
-        elif row['hostname'] and not row['url']:
-            return row['application'] + '://' + row['hostname']
+    if hostname and path:
```

Comment:
The columns for the URL can be empty. Do you know if this means that the URL is unknown or is it equivalent with `/`?

Reply:
I checked the Shadowserver reports for the past weeks.
If "url" is present (and does not include the full HTTP request :-/), it always starts with "/".
There are also lots of events with a url of only "/".
So I consider an empty value as the url not provided at all.